pkg/network/master/egressip.go

```diff
@@ -230,10 +287,60 @@ func (eim *egressIPManager) check(retrying bool) (bool, error) {
 func (eim *egressIPManager) Synced() {
 }
 
+// ClaimEgressIP will create the CloudPrivateIPConfig object. There is one
+// tricky condition which we will need to handle, namely: when moving one egress
+// IP from node A to node B. In that event: we will process the removal from
+// node A before the add to node B, this can however take a bit of time (namely:
+// the time taken for the cloud controller to process and execute the event and
+// the cloud API to perform the action). Given that CloudPrivateIPConfig is
+// keyed by IP address: we won't be able to create the assignment to node B
+// until the previous assignment to node A has been fully removed. We thus need
```

Comment:
why can't we just Update the CloudPrivateIPConfig rather than doing delete+add?

Reply:
Mainly because I didn't want to refactor the existing implementation of egress IP too broadly given the lack of E2E test coverage. The current implementation performs atomic ADD/DEL operations on a per-egress-IP-to-node basis (because nodes never had to update the egress IP setup: it either added, removed it or did both), and this fits that mode without refactoring the current logic.

Reply:
> Mainly because I didn't want to refactor the existing implementation of egress IP too broadly given the lack of E2E test coverage.

yeah. sigh.